tests/shardddl2/run.sh

```diff
@@ -0,0 +1,715 @@
+}
+
+function DM_065_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} modify a bigint after b;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} modify a bigint first;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} modify a bigint first;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9);"
+
+    if [[ "$1" = "pessimistic" ]]; then
+        check_log_contain_with_retry "is different with" $WORK_DIR/master/log/dm-master.log
+    else
+        run_sql_tidb_with_retry "select count(1) from ${shardddl}.${tb};" "count(1): 9"
+    fi
+}
+
+function DM_065() {
+    run_case 065 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int);\"" \
+    "clean_table" "pessimistic"
+    run_case 065 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int);\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_066_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} modify id int default 1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} modify id int default 1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} modify id int default 1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9);"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+function DM_066() {
+    run_case 066 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    # currently not support optimistic
+    # run_case 066 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
```

Comment:
why not support optimistic?

Reply:
I forgot to remove the comment. It's supported in c2bd654